communication/mavlink_stream.c

```diff
@@ -53,20 +53,33 @@
 //------------------------------------------------------------------------------
 
 void mavlink_stream_init(	mavlink_stream_t* mavlink_stream, 
-							const mavlink_stream_conf_t* config, 
-							byte_stream_t* rx_stream, 
-							byte_stream_t* tx_stream)
-{	
-	// mavlink_tx_stream                 = tx_stream;
-	mavlink_stream->tx         		  = tx_stream;
-	mavlink_stream->rx                = rx_stream;
-	mavlink_stream->sysid             = config->sysid;
-	mavlink_stream->compid            = config->compid;
-	mavlink_stream->use_dma			  = config->use_dma;
-	mavlink_stream->msg_available     = false;
+				const mavlink_stream_conf_t* config, 
+				byte_stream_t* rx_stream, 
+				byte_stream_t* tx_stream)
+{
+	// Init static variable storing number of mavlink stream instances
+	static uint8_t nb_mavlink_stream_instances = 0;
+	if( nb_mavlink_stream_instances < MAVLINK_COMM_NUM_BUFFERS )
+	{
+		mavlink_stream->mavlink_channel =  nb_mavlink_stream_instances;
```

Comment:
Could you use the MAVLink enum with a switch case? It would be more transparent for standard MAVLink users